explorer/ui/app/components/details/From.tsx

```diff
@@ -65,7 +66,18 @@ export const From = ({ xMsgDetails }) => {
           onClick={() => copyToClipboard(xMsgDetails?.txHash)}
         />
       </div>
-      {/* Offset */}
+      {/* Block Timestamp */}
+      <div className="flex mt-5 pb-2 border-b-[1px] border-subtle border-solid">
+        <p className="w-[150px] sm:w-48 text-sm">Block Timestamp</p>
+        <p className="flex-1">
+          <span className="text-default">
+            {dateFormatter(new Date(xMsgDetails?.block.timestamp))}{' '}
+          </span>
+         ( {xMsgDetails?.block.timestamp &&
+            dateFormatterXMsgPage(new Date(xMsgDetails?.block.timestamp))})
```

Comment:
Once you check for undefefined / null, you dont' have to check again

Reply:
```suggestion
         ( {xMsgDetails?.block.timestamp &&
            dateFormatterXMsgPage(new Date(xMsgDetails.block.timestamp))})
```

Reply:
Also nit but I tend to[ prefer ternaries over &&](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx)
```suggestion
         ( {xMsgDetails?.block.timestamp ? 
            dateFormatterXMsgPage(new Date(xMsgDetails?.block.timestamp)) : null})
```